src/Input/Parser.re

```diff
@@ -1,50 +1,69 @@
-let dMinus = Str.regexp_case_fold(".*d-.*");
-let sMinus = Str.regexp_case_fold(".*s-.*");
+module Regexes = {
+  let control = Str.regexp_case_fold(".*c-.*");
+  let alt = Str.regexp_case_fold(".*a-.*");
+  let command = Str.regexp_case_fold(".*d-.*");
+  let shift = Str.regexp_case_fold(".*s-.*");
+  let compound = Str.regexp_case_fold({|<\(.-\)+\(.*\)>|});
+  let bracketized = Str.regexp_case_fold({|<\(.*\)>|});
+};
+
+let bracketize = key =>
+  if (String.length(key) > 1) {
+    "<" ++ key ++ ">";
+  } else {
+    key;
+  };
 
 /*
  [toFriendlyName(key)] takes a vim-style key description,
  like ["<C-v>"], and returns a friendly representation, like ["Ctrl+v"]
  */
-let toFriendlyName = (v: string) => {
-  let len = String.length(v);
-
-  switch (v) {
-  | " " => Some("Space")
-  | "<BS>" => Some("Backspace")
-  | "<ESC>" => Some("Escape")
-  | "<TAB>" => Some("Tab")
-  | "<C-TAB>" => Some("Control + Tab")
-  | "<CR>" => Some("Enter")
-  | _ =>
-    if (len == 0) {
-      None;
-    } else if (len == 1) {
-      Some(v);
-    } else {
-      let firstCharacter = v.[0];
-      let lastCharacter = v.[len - 1];
-
-      if (firstCharacter == '<' && lastCharacter == '>') {
-        let hasControl = Str.string_match(cMinus, v, 0);
-        let hasShift = Str.string_match(sMinus, v, 0);
-        let hasAlt = Str.string_match(aMinus, v, 0);
-        let hasCommand = Str.string_match(dMinus, v, 0);
-
-        let character = v.[len - 2];
-
-        let ret = hasCommand ? "Command + " : "";
-        let ret = ret ++ (hasControl ? "Control + " : "");
-        let ret = ret ++ (hasAlt ? "Alt + " : "");
-
-        let character =
-          !hasShift ? Char.lowercase_ascii(character) : character;
-
-        Some(ret ++ String.make(1, character));
+let rec toFriendlyName =
+  fun
+  | " " => "Space"
+  | "<BS>" => "Backspace"
+  | "<ESC>" => "Escape"
+  | "<CR>" => "Enter"
+  | key when String.length(key) == 0 => ""
+  | key when String.length(key) == 1 => key
+  | key => {
+      let isCompound = Str.string_match(Regexes.compound, key, 0);
+
+      if (isCompound) {
+        let buffer = Buffer.create(String.length(key));
+        let significand =
+          Str.matched_group(2, key) |> bracketize |> toFriendlyName;
+
+        let hasControl = Str.string_match(Regexes.control, key, 0);
+        let hasShift = Str.string_match(Regexes.shift, key, 0);
+        let hasAlt = Str.string_match(Regexes.alt, key, 0);
+        let hasCommand = Str.string_match(Regexes.command, key, 0);
+
+        if (hasCommand) {
+          Buffer.add_string(buffer, "Command + ");
+        };
+        if (hasControl) {
+          Buffer.add_string(buffer, "Control + ");
+        };
+        if (hasAlt) {
+          Buffer.add_string(buffer, "Alt + ");
+        };
+        if (hasShift) {
+          Buffer.add_string(buffer, "Shift + ");
+        };
+
+        Buffer.add_string(buffer, significand);
+
+        Buffer.contents(buffer);
```

Comment:
@glennsl This actually might be `command` for mac, `Windows` or `Super` for Linux.
Best would be to have different names for all OS's but `Super` might fit all if not doable right now
(for ex. `Command` will hurt me a lot since I didn't use to work on Mac, it going to be misread as `Control`

Also would be better to have the save order of control keys as on:
https://github.com/onivim/oni2/blob/master/src/Input/Handler.re#L85

Reply:
Yeah, good point. We could switch this based on OS. But is the Windows key every spelled out entirely, or is it always just `Win`?

I think we should keep the ordering though, as this is much more natural. But perhaps we should change the order in `Handler`. I don't know if the order is significant for vim though.

Reply:
And Done.